internal/dag/dag.go

```diff
@@ -424,14 +457,30 @@ func (d *DAG) processIngressRoute(ir *ingressroutev1.IngressRoute, prefixMatch s
 				// we are delegating to another IngressRoute in the same namespace
 				namespace = ir.Namespace
 			}
-			dir, ok := d.ingressroutes[meta{name: route.Delegate.Name, namespace: namespace}]
+			dest, ok := d.ingressroutes[meta{name: route.Delegate.Name, namespace: namespace}]
 			if ok {
+				// dest is not an orphaned route, as there is an IR that points to it
+				orphaned[meta{name: dest.Name, namespace: dest.Namespace}] = false
+
+				// ensure we are not following an edge that produces a cycle
+				var path []string
+				for _, vir := range visited {
+					path = append(path, fmt.Sprintf("%s/%s", vir.Namespace, vir.Name))
+				}
+				for _, vir := range visited {
```

Comment:
this makes me think that visited should probably be a map, not a slice. You're looking for in the set of visited elements to see if we've already passed this way before. The order is not important, only set membership. 

A map might be a better choice for visited and would make my comments above mute.

Reply:
We were using a map but switched to a slice so that we can build up the delegation path and include it in the error msg in case of a cycle. There's more context in the "Cycle Detection" section of the PR description.